view/src/components/foodbanks.js

```diff
@@ -0,0 +1,926 @@
+import React, { Component } from "react";
+
+import Address from "../extras/address";
+
+import withStyles from "@material-ui/core/styles/withStyles";
+import Typography from "@material-ui/core/Typography";
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import AddCircleIcon from "@material-ui/icons/AddCircle";
+import AppBar from "@material-ui/core/AppBar";
+import Toolbar from "@material-ui/core/Toolbar";
+import IconButton from "@material-ui/core/IconButton";
+import CloseIcon from "@material-ui/icons/Close";
+import Slide from "@material-ui/core/Slide";
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import Card from "@material-ui/core/Card";
+import Container from "@material-ui/core/Container";
+import CardActions from "@material-ui/core/CardActions";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import CardContent from "@material-ui/core/CardContent";
+import MuiDialogTitle from "@material-ui/core/DialogTitle";
+import MuiDialogContent from "@material-ui/core/DialogContent";
+import InputBase from "@material-ui/core/InputBase";
+import Chip from "@material-ui/core/Chip";
+import SearchIcon from "@material-ui/icons/Search";
+import InputAdornment from "@material-ui/core/InputAdornment";
+import Box from "@material-ui/core/Box";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import InputLabel from "@material-ui/core/InputLabel";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+import Checkbox from "@material-ui/core/Checkbox";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import PropTypes from "prop-types";
+import MaskedInput from "react-text-mask";
+import OutlinedInput from "@material-ui/core/OutlinedInput";
+
+import axios from "axios";
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+import { authMiddleWare } from "../util/auth";
+
+const TAG_EXAMPLES = [
+  { title: "High Food Insecurity" },
+  { title: "Major City" },
+];
+
+const styles = (theme) => ({
+  content: {
+    flexGrow: 1,
+    padding: theme.spacing(3),
+  },
+  appBar: {
+    position: "relative",
+  },
+  title: {
+    marginLeft: theme.spacing(2),
+    flex: 1,
+  },
+  submitButton: {
+    display: "block",
+    color: "white",
+    textAlign: "center",
+    position: "absolute",
+    top: 14,
+    right: 10,
+  },
+  floatingButton: {
+    position: "fixed",
+    bottom: 0,
+    right: 0,
+  },
+  form: {
+    width: "98%",
```

Comment:
Percentage-based widths that are close to, but slightly less than, 100% are recipes for unexpected scrollbars in my experience, because at some width the margin will get smaller than you expect.  One thing you can do here is use `calc()`, which lets you say something like `calc(100% - 32px)`.